### PR TITLE
Avoid division by zero in tensordot, allowing 0-length arrays

### PR DESCRIPTION
**Pull request by pentschev**

### CODE REVIEW

**Comment by pentschev** [comment]
It looks like AppVeyor didn't run at all. Could someone retrigger the build to check this?

**Comment by toslunar** [comment]
Jenkins, test this please.

File: cupy/linalg/product.py [hunk 1]
```diff
@@ -189,8 +189,9 @@ def tensordot(a, b, axes=2):
     ret_shape = a.shape[sum_ndim:] + b.shape[sum_ndim:]
 
     k = internal.prod(a.shape[:sum_ndim])
-    n = a.size // k
-    m = b.size // k
+    # Avoid division by zero, allowing 0-length dimensions to work
```
[the remark this answers is not in the document]
Comment (pentschev) [reply cupy/linalg/product.py]: Sure, did that now. Please let me know if this is satisfactory.

File: tests/cupy_tests/linalg_tests/test_product.py [hunk 2]
```diff
@@ -349,6 +349,27 @@ def test_zerodim_kron(self, xp, dtype):
...
+    'params': [
+        ((0, 0), 2),
+        ((0, 0), (1, 0)),
+        ((0, 0, 0), 2),
+        ((0, 0, 0), 3),
+        ((0, 0, 0), ([2, 1], [0, 2])),
+        ((0, 0, 0), ([0, 2, 1], [1, 2, 0])),
+    ],
+}))
+@testing.gpu
```
[the remark this answers is not in the document]
Comment (pentschev) [reply tests/cupy_tests/linalg_tests/test_product.py]: I wasn't aware of that, thanks for letting me know. Fixed that now.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by pentschev** [comment]
Thanks for the review @toslunar!

**Comment by pentschev** [comment]
Thanks for merging @toslunar!

**Pull request merged**